source/gtd/GirWrapper.d

```diff
@@ -585,56 +588,56 @@ class GirWrapper
 		}
 	}
 
-	private string getGirDirectory()
+	private string[] getGirDirectories()
 	{
-		version(Windows)
+		static string[] dirs;
+		if(dirs !is null)
 		{
-			import std.process : environment;
-
-			static string path;
+			return dirs;
+		}
+		else
+		{
+			dirs = [];
+		}
 
-			if (path !is null)
-				return path;
+		if ( commandlineGirPath )
+		{
+			dirs ~= commandlineGirPath;
+		}
 
+		version(Windows)
+		{
 			foreach (p; splitter(environment.get("PATH"), ';'))
 			{
 				string dllPath = buildNormalizedPath(p, "libgtk-3-0.dll");
 
-				if ( exists(dllPath) )
-					path = p.buildNormalizedPath("../share/gir-1.0");
+				dirs ~= p.buildNormalizedPath("../share/gir-1.0");
 			}
 
-			return path;
 		}
 		else version(OSX)
 		{
-			import std.process : environment;
-
-			static string path;
-
-			if (path !is null)
-				return path;
-
-			path = environment.get("GTK_BASEPATH");
+			string path = environment.get("GTK_BASEPATH");
 			if(path)
 			{
-				path = path.buildNormalizedPath("../share/gir-1.0");
+				dirs ~= path.buildNormalizedPath("../share/gir-1.0");
 			}
-			else
-			{
-				path = environment.get("HOMEBREW_ROOT");
-				if(path)
+			path = environment.get("HOMEBREW_ROOT");
+			if(path)
 				{
-					path = path.buildNormalizedPath("share/gir-1.0");
+					dirs ~= path.buildNormalizedPath("share/gir-1.0");
 				}
-			}
-
-			return path;
 		}
 		else
 		{
-			return "/usr/share/gir-1.0";
+			string xdgDataDirs = environment.get("XDG_DATA_DIRS", "/usr/local/share:/usr/share");
+		        foreach (p; splitter(xdgDataDirs, ':'))
+		        {
+		        	dirs ~= p.buildNormalizedPath("gir-1.0");
```

Comment:
Should we check if the path exists before appending it to the result?

Reply:
I could also replace this with a filter before the return that removes the directories that don't exist. I wanted to have the caller responsible for checking if the dirs actually exist, but adding a check to make sure the path actually exists may be a good idea